src/Entity/OgMembership.php

```diff
@@ -8,35 +8,42 @@
 namespace Drupal\og\Entity;
 
 use Drupal\Core\Entity\ContentEntityBase;
-use Drupal\Core\Entity\ContentEntityInterface;
-use Drupal\Core\Entity\EntityInterface;
 use Drupal\Core\Entity\EntityStorageInterface;
 use Drupal\Core\Entity\EntityTypeInterface;
 use Drupal\Core\Field\BaseFieldDefinition;
 use Drupal\Core\Field\FieldStorageDefinitionInterface;
-use Drupal\Core\TypedData\DataDefinition;
 use Drupal\og\OgGroupAudienceHelper;
+use Drupal\og\OgMembershipInterface;
 
 /**
- * The OG membership is the main idea behind OG. The OG membership entity keep
- * the connection between the group and the her content. For example we have the
- * node 1 which is a group and the node 2 which is node that belong to a group:
+ * While referencing content to a group done using the normal entity storage in
+ * the case of a user we have couple of things we need to handle. When a user is
+ * a member of a group we need to know if the membership is active, pending or
+ * blocking. Another use case we need to cover is the roles the user own in a
+ * specific group: member, content editor, manager etc. etc.
+ *
+ * This is the main reason we need the OG membership. The OG membership entity
+ * keeps the connection between the group and the user. For example we have node
+ * 1 which is a group and user 2 which is a user that belong to a group:
```

Comment:
10x. I really tried to explain this in the best way.

BTW - are you coming to dev days 2016?